### PR TITLE
Teleport: Update Rules

### PR DESCRIPTION
**Pull request by jof**
This PR updates the Gravitational Teleport rule set.

- The documentation links for existing rules needed some updating to new locations
- Added the name of the Teleport cluster into some of titles (useful for multi-cluster organizations)
- Added a Rule to detect local user logins without using multi-factor auth
- Added a Rule to detect the creation of Teleport Locks (used to pre-emptively disable previously-issued certificates)
- Added a Rule to detect the creation of abnormally long-lived certificates
- Added a Rule to detect when Roles are created
- Added a Rule to detect logins as root
- Added a Rule to detect the creation of SAML connectors

### CODE REVIEW

File: global_helpers/panther_default.py [hunk 1]
```diff
@@ -125,3 +125,5 @@ def aws_regions() -> List[str]:
         "us-west-1",
         "us-west-2",
     ]
+
```
Comment (arielkr256) [inline global_helpers/panther_default.py:128]:
```suggestion
```

File: global_helpers/panther_default.py [hunk 2]
```diff
@@ -125,3 +125,5 @@ def aws_regions() -> List[str]:
...
+
```
Comment (arielkr256) [inline global_helpers/panther_default.py:129]:
```suggestion
```

File: global_helpers/panther_default.py [hunk 3]
```diff
@@ -125,3 +125,4 @@ def aws_regions() -> List[str]:
         "us-west-1",
         "us-west-2",
     ]
+
```
Comment (arielkr256) [inline global_helpers/panther_default.py:128]:
```suggestion
```

File: rules/gravitational_teleport_rules/teleport_long_lived_certs.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    #     "other_example_role": (timedelta(hours=720), datetime.max),
+    # },
+}
+
+
+def rule(event):
+    if not event.get("event") == "cert.create":
+        return False
+    max_validity = MAXIMUM_NORMAL_VALIDITY_INTERVAL + ISSUANCE_GRACE_PERIOD
+    for role in event.get("identity", {}).get("roles", []):
```
Comment (arielkr256) [inline rules/gravitational_teleport_rules/teleport_long_lived_certs.py:51]:
```suggestion
    for role in event.deep_get("identity", "roles", default=[]):
```

File: rules/gravitational_teleport_rules/teleport_long_lived_certs.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        if validity and expiration:
+            # Ignore exceptions that have passed their expiry date
+            if datetime.utcnow() < expiration:
+                max_validity = max(max_validity, validity)
+    return validity_interval(event) > max_validity
+
+
+def validity_interval(event):
+    event_time = panther_nanotime_to_python_datetime(event.get("time"))
+    expires = golang_nanotime_to_python_datetime(event.get("identity", {}).get("expires"))
```
Comment (arielkr256) [inline rules/gravitational_teleport_rules/teleport_long_lived_certs.py:64]:
```suggestion
    expires = golang_nanotime_to_python_datetime(event.deep_get("identity", "expires", default=None))
```

File: rules/gravitational_teleport_rules/teleport_long_lived_certs.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    if not event_time and expires:
+        return False
+    interval = expires - event_time
+    return interval
+
+
+def title(event):
+    return (
+        f"A Certificate for  [{event.get('identity', {}).get('user', '<Cert with no User!?>')}] "
```
Comment (arielkr256) [inline rules/gravitational_teleport_rules/teleport_long_lived_certs.py:73]:
```suggestion
        f"A Certificate for  [{event.deep_get('identity', 'user', default='<Cert with no User!?>')}] "
```

File: packs/gravitational_teleport.yml [hunk 7]
```diff
@@ -10,4 +10,5 @@ PackDefinition:
     - Teleport.SuspiciousCommands
     # Globals used in these detections
     - panther_base_helpers
+    - panther_default
```
Comment (arielkr256) [inline packs/gravitational_teleport.yml:13]: Is this used?

**Review by arielkr256 — COMMENTED** [review]
Overall looks great!  Made a couple of suggestions to use `event.deep_get()` instead of chained gets.  https://github.com/panther-labs/panther-analysis/blob/main/global_helpers/panther_base_helpers.py#L301

File: packs/gravitational_teleport.yml (hunk 7 above)
Comment (jof) [reply packs/gravitational_teleport.yml:13]: Oops, no. I had some helpers in `panther_default`, but I moved them to `panther_base_helpers`.

**Comment by jof** [comment]
@arielkr256 -- good call on the `.deep_get()` access style; way more readable. Added your suggestions.

File: rules/gravitational_teleport_rules/teleport_long_lived_certs.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        if validity and expiration:
+            # Ignore exceptions that have passed their expiry date
+            if datetime.utcnow() < expiration:
+                max_validity = max(max_validity, validity)
+    return validity_interval(event) > max_validity
+
+
+def validity_interval(event):
+    event_time = panther_nanotime_to_python_datetime(event.get("time"))
+    expires = golang_nanotime_to_python_datetime(event.deep_get("identity", "expires", default=None))
```
Comment (arielkr256) [inline rules/gravitational_teleport_rules/teleport_long_lived_certs.py:64]:
```suggestion
    expires = golang_nanotime_to_python_datetime(
        event.deep_get("identity", "expires", default=None)
        )
```

File: rules/gravitational_teleport_rules/teleport_long_lived_certs.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    if not event_time and expires:
+        return False
+    interval = expires - event_time
+    return interval
+
+
+def title(event):
+    return (
```
Comment (arielkr256) [inline rules/gravitational_teleport_rules/teleport_long_lived_certs.py:72]:
```suggestion
    identity = event.deep_get('identity', 'user', default='<Cert with no User!?>')
    return (
```

**Pull request merged**